src/keyring/KeyringController.ts

```diff
@@ -442,6 +445,16 @@ export class KeyringController extends BaseController<
   ) {
     try {
       const address = normalizeAddress(messageParams.from);
+
+      const ledgerKeyring = await this.getLedgerKeyring();
+      const isLedgerAccount = await ledgerKeyring.managesAccount(address);
+
+      if (isLedgerAccount) {
+        return privates
+          .get(this)
+          .keyring.signTypedMessage(messageParams, { version });
+      }
```

Comment:
For a next iteration we could create methods to hold each block of code (Ledger, Keystone, etc)

Reply:
Its not necessary to address it in this PR

Reply:
Ideally we should push down the code into the respective keyrings as the "account check" logic is already performed by the Ethereum keyring controller. However even for the MM code the signTypedData logic lives in the app and not in the keyring.